fedimint-api/src/module/mod.rs

```diff
@@ -319,6 +360,16 @@ pub trait ModuleGen: Debug + Sized {
 
     fn decoder(&self) -> Self::Decoder;
 
+    /// Version of the consensus supported by this implementation
+    ///
+    /// Refer to `ConsensuVersion` for more information about versioning.
+    ///
+    /// One module implementation ([`ModuleGen`] of a given [`ModuleKind`]) can potentially
+    /// implement multiple versions of the consensus, and depending on the config module instance
+    /// config, instantiate the desired one. This method should should expose all the available
+    /// versions, purely for information, setup UI and sanity checking purposes.
+    fn versions(&self) -> &[ConsensusVersion];
```

Comment:
I'd strongly recommend against the added complexity of supporting different consensus versions in the same module code base. The same module can be imported multiple times at different versions if needed, but each would only be at one version.